### PR TITLE
Fixes to netdata.spec.in to build on CentOS/RHEL

### PR DESCRIPTION
**Pull request by sketch242**
Tested on CentOS 6 and 7.

Not sure if the last %if/%else is even necessary, or if the existing spec is broken everywhere currently.  Are there any RPM-based distros where the binary would be built in (/usr)/bin instead of (/usr)/sbin?

### CODE REVIEW

**Comment by ktsaou** [comment]
@alonbl @philwhineray can you review this?

File: netdata.spec.in [hunk 1]
```diff
@@ -7,11 +7,13 @@ URL:		http://firehol.org
 Source:		%{name}-@PACKAGE_VERSION@.tar.bz2
 
 BuildRequires:	libmnl-devel
-BuildRequires:	libnetfilter_acct-devel
 BuildRequires:	zlib-devel
 Requires:	libmnl
-Requires:	libnetfilter_acct
 Requires:	zlib
+%if !0%{?rhel}
```
Comment (alonbl) [inline netdata.spec.in]: I can only guess that rhel-7 does have this package, can you confirm it do not.

File: netdata.spec.in [hunk 2]
```diff
@@ -47,18 +47,25 @@ make %{?_smp_mflags}
 rm -rf "%{buildroot}"
 make %{?_smp_mflags} install DESTDIR="%{buildroot}"
 find "%{buildroot}" -name .keep -exec rm {} \;
+touch %{buildroot}/%{_sysconfdir}/netdata/netdata.conf
```
Comment (alonbl) [inline netdata.spec.in]: why is this required? it will just create conflicts.

File: netdata.spec.in [hunk 3]
```diff
@@ -47,18 +47,25 @@ make %{?_smp_mflags}
...
+%if 0%{?fedora} || 0%{?rhel} >= 7 
+install -m 0644 -D system/netdata-systemd ${RPM_BUILD_ROOT}%{_unitdir}/netdata.service
+%endif
```
Comment (alonbl) [inline netdata.spec.in]: please use systemd macros to install units, you should also add a %pre and %post to properly handle systemd. please refer to fedora documentation.

File: netdata.spec.in [hunk 4]
```diff
@@ -47,18 +47,25 @@ make %{?_smp_mflags}
...
 %attr(-, netdata, netdata) %dir %{_localstatedir}/cache/%{name}/
 %attr(-, netdata, netdata) %dir %{_localstatedir}/log/%{name}/
+%attr(-, netdata, netdata) %{_datadir}/%{name}/
 %config(noreplace) %{_sysconfdir}/%{name}/
 %if 0%{?rhel}
 %{_sbindir}/%{name}
 %else
 %{_bindir}/%{name}
 %endif
-%{_datadir}/%{name}/
```
Comment (alonbl) [inline netdata.spec.in]: why have you removed this? does it missing from installation?

File: netdata.spec.in (hunk 1 above)
Comment (sketch242) [inline netdata.spec.in]: libnetfilter_acct does not exist on any current release of RHEL (up to 7.2).

File: netdata.spec.in (hunk 4 above)
Comment (sketch242) [inline netdata.spec.in]: I replaced it with line 58, which is the same path, but with permissions that allow netdata to run as the user netdata, instead of requiring root.

File: netdata.spec.in (hunk 2 above)
Comment (sketch242) [inline netdata.spec.in]: My preference for packages is that they will run with a default config out of the box.  As it currently builds, netdata won't start because there is no config file.  There isnt' even a default config in the release tarball at all, so I just create an empty file and let it start up with the defaults.

Comment (alonbl) [inline netdata.spec.in]: application should start with no file, if it is not the case we should fix the application.

File: netdata.spec.in (hunk 4 above)
Comment (alonbl) [inline netdata.spec.in]: /usr should not be modified by the package, if there is a need to modify anything post installation it should go into /var/lib

File: netdata.spec.in [hunk 5]
```diff
@@ -48,7 +52,11 @@ find "%{buildroot}" -name .keep -exec rm {} \;
 %attr(-, netdata, netdata) %dir %{_localstatedir}/cache/%{name}/
 %attr(-, netdata, netdata) %dir %{_localstatedir}/log/%{name}/
 %config(noreplace) %{_sysconfdir}/%{name}/
+%if 0%{?rhel}
+%{_sbindir}/%{name}
```
Comment (alonbl) [inline netdata.spec.in]: there should be no difference in file locations between rhel and fedora.

File: netdata.spec.in (hunk 1 above)
Comment (alonbl) [inline netdata.spec.in]: ok

File: netdata.spec.in (hunk 4 above)
Comment (sketch242) [inline netdata.spec.in]: Not sure we are talking about the same thing here. I was referring to line 58 on the right side, in the full patch, which is not shown in the snippet above (which starts at 59).  It sets the ownership of /usr/share/netdata to netdata:netdata.  This is required by the application when running as user netdata.  If it's owned by root, the application will log an error and refuse to read the dir, even though permissions permit read access:

16-03-31 07:33:39: ERROR: netdata: 1: File '/usr/share/netdata/web/' is owned by user 0 (I run as user 498). Access Denied.

I only added the if/else for rhel on bindir/sbindir because that's what I was fixing at the time.  That should be the case for both fedora and rhel, though. I thought about removing sbindir entirely, but wasn't sure if it might break other rpm-based distros like SUSE, so if %if'd it instead.

File: netdata.spec.in (hunk 3 above)
Comment (sketch242) [inline netdata.spec.in]: I'm new to github, is it better to modify the existing patch (if that's even still possible), or create a new one at this point?

Comment (alonbl) [inline netdata.spec.in]: you should push with --force to the same branch.

File: netdata.spec.in (hunk 4 above)
Comment (alonbl) [inline netdata.spec.in]: @ktsaou: this is bad application behavior, please move all files that are updated to localstatedir (/var/lib)

Comment (ktsaou) [inline netdata.spec.in]: You suggest to move web files to `/var/lib/netdata/web`?
This is easy and is controlled by the make files (I think). The `netdata-installer.sh` has a default that is used only when the configuration file is not available. It does not set it during installation. I think this is taken from the make system.

Why don't you include this change here?

Comment (alonbl) [inline netdata.spec.in]: I do not follow. the netdata-installer.sh was always align to packaging. in best case it should copy the /usr/share files and create an instance at /var/lib to be used. otherwise, you should use the static data from /usr/share and only store the files that you write into in /var/lib, if you must have one directory for all, create /var/lib and symlink /usr/share for static data.

Comment (ktsaou) [inline netdata.spec.in]: ok. We are talking about different things.

The path `/usr/share/netdata/web` is never written by netdata. Its contents need to be owned by netdata though. Why? netdata is a web server itself. To protect the system, this web server refuses to read files that are not owned by the user the process runs.

netdata currently writes files only at `/var/cache/netdata` and `/var/log/netdata`.
Nowhere else.

Comment (ktsaou) [inline netdata.spec.in]: So, if you want to move the web files from `/usr/share/netdata/web` to `/var/lib/netdata/web`, we can do it.
Is it really needed?
(I'll have to update the installer to re-write the netdata configuration to fix this path - since this change will introduce a problem to all those that have it already installed).

Comment (alonbl) [inline netdata.spec.in]: if netdata does not write to /usr/share this change is not required, back to my initial comment.

**Comment by ktsaou** [comment]
I am planning for a new release tomorrow. Shall we include this?

**Comment by philwhineray** [comment]
I get the impression this might not quite be ready yet?  I cannot claim any experience with redhat spec files so I am going only on the comments above.

Anyway I have added a netdata team, given it write access to the repo and added @alonbl to it; this will enable him to issue a pull direct if/when satisfied the changes are an improvement to the original.

**Comment by alonbl** [comment]
I cannot understand the integrity of last push, please see #140 as a superset.
